src/constants/sinsemilla.rs

```diff
@@ -0,0 +1,255 @@
+//! Sinsemilla generators
+use super::{OrchardFixedBases, OrchardFixedBasesFull};
+use crate::circuit::gadget::sinsemilla::{CommitDomains, HashDomains};
+use crate::spec::i2lebsp;
+
+use pasta_curves::{
+    arithmetic::{CurveAffine, FieldExt},
+    pallas,
+};
+
+/// Number of bits of each message piece in $\mathsf{SinsemillaHashToPoint}$
+pub const K: usize = 10;
+
+/// $\frac{1}{2^K}$
+pub const INV_TWO_POW_K: [u8; 32] = [
+    1, 0, 192, 196, 160, 229, 70, 82, 221, 165, 74, 202, 85, 7, 62, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 240, 63,
+];
+
+/// The largest integer such that $2^c \leq (r_P - 1) / 2$, where $r_P$ is the order
+/// of Pallas.
+pub const C: usize = 253;
+
+/// $\ell^\mathsf{Orchard}_\mathsf{Merkle}$
+pub(crate) const L_ORCHARD_MERKLE: usize = 255;
+
+/// SWU hash-to-curve personalization for the Merkle CRH generator
+pub const MERKLE_CRH_PERSONALIZATION: &str = "z.cash:Orchard-MerkleCRH";
+
+// Sinsemilla Q generators
+
+/// SWU hash-to-curve personalization for Sinsemilla $Q$ generators.
+pub const Q_PERSONALIZATION: &str = "z.cash:SinsemillaQ";
+
+// Sinsemilla S generators
+
+/// SWU hash-to-curve personalization for Sinsemilla $S$ generators.
+pub const S_PERSONALIZATION: &str = "z.cash:SinsemillaS";
```

Comment:
These constants are part of the definition of Sinsemilla, so these constants should end up wherever the Sinsemilla chip does (i.e. in `halo2_gadgets`). Meanwhile `MERKLE_CRH_PERSONALIZATION`, and the hard-coded generators below, are Orchard-specific and will stick around here.

Reply:
I'll import these from `src/primitives/sinsemilla`.